fix(service-worker): correctly handle failed cache-busted request

gkalpak opened the pull request:
Since 5be4edfa17120ab1b9c9e8d87b7b539cdd180a25, a failing cache-busted network request (such as requests for fetching an uncached asset) will cause the ServiceWorker to incorrectly enter a degraded `EXISTING_CLIENTS_ONLY` mode. A failing network request could be caused by many reasons, including the client or server being offline, and does not necessarily signify a broken ServiceWorker state.

This commit fixes the logic in `cacheBustedFetchFromNetwork()` to correctly handle errors in network requests.
For more details on the problem and the implemented fix see #39775.

Fixes #39775.

mary-poppins commented:
You can preview d834d24 at https://pr39786-d834d24.ngbuilds.io/.

sonukapoor commented on packages/service-worker/worker/src/assets.ts:
```diff
@@ -388,18 +388,17 @@ export abstract class AssetGroup {
-      if (networkResult.ok) {
+      let response = await this.safeFetch(req);
+
+      // Decide whether a cache-busted request is necessary. A cache-busted request is necessary iff
```
```suggestion
      // Decide whether a cache-busted request is necessary. A cache-busted request is necessary if
```

gkalpak replied:
Well, this was [intended][1], but I'll change it since it might confuse some people 😅
Thx for taking a look ✨ 

[1]: https://en.wikipedia.org/wiki/If_and_only_if

mary-poppins commented:
You can preview 31f8591 at https://pr39786-31f8591.ngbuilds.io/.

gkalpak committed:
fix(service-worker): correctly handle failed cache-busted request

Since 5be4edfa17120ab1b9c9e8d87b7b539cdd180a25, a failing cache-busted
network request (such as requests for fetching uncached assets) will
cause the ServiceWorker to incorrectly enter a degraded
`EXISTING_CLIENTS_ONLY` mode. A failing network request could be caused
by many reasons, including the client or server being offline, and does
not necessarily signify a broken ServiceWorker state.

This commit fixes the logic in `cacheBustedFetchFromNetwork()` to
correctly handle errors in network requests.
For more details on the problem and the implemented fix see #39775.

Fixes #39775

gkalpak committed:
fixup! fix(service-worker): correctly handle failed cache-busted request

mary-poppins commented:
You can preview febee11 at https://pr39786-febee11.ngbuilds.io/.

IgorMinar reviewed (approved):
LGTM
